cpp/include/raft/sparse/hierarchy/detail/mst.cuh

```diff
@@ -201,6 +201,12 @@ void build_sorted_mst(const raft::handle_t &handle, const value_t *X,
                " or increase 'max_iter'",
                max_iter);
 
+  RAFT_EXPECTS(mst_coo.n_edges == m - 1,
```

Comment:
I think you meant `2m-2 `

Reply:
It's a de-symmetrized output, or supposed to be at least

Reply:
In the case of SLHC, we don't symmetrize the output. Divye had added an argument to turn off symmetrized output in 0.19.

Reply:
Based on the wording I think I should clarify that symmetrize was not a post-processing step. I haven't looked at the code I don't think one can just skip safely skip this as you may run into propagation issues

Reply:
The propagation part isn't affected. `symmetrize=false` only adds the lower colored src->dst pair to the output that is copied as a COO, but that part is not used in the color propagation code

Reply:
@afender, thanks for the quick review on this one! My apologies for the delayed reply. This week, I'm hoping to do some validation on this in order to root cause https://github.com/rapidsai/cuml/issues/3741. Hopefully I'll get some more clues as to what's going on.